src/time.rs

```diff
@@ -0,0 +1,278 @@
+    ) -> Self {
+        let t = Self::combine_parts(low_bits, high_bits);
+        Self::new_truncate(t)
+    }
+
+    const fn combine_parts(low_bits: NanosecondsLowBits, high_bits: NanosecondsHighBits) -> u64 {
+        let low = low_bits.0.to_le_bytes();
+        let high = high_bits.0.to_le_bytes();
+        u64::from_le_bytes([
+            low[0], low[1], low[2], low[3], high[0], high[1], high[2], high[3],
+        ])
+    }
+
+    /// Get the underlying value as a convenient primitive
+    pub const fn get(&self) -> u64 {
+        self.0
+    }
+
+    /// Split time into little endian parts.
+    pub fn split(self) -> (NanosecondsLowBits, NanosecondsHighBits) {
+        let le_bytes = self.0.to_le_bytes();
+
+        let low_bits = NanosecondsLowBits(u32::from_le_bytes([
+            le_bytes[0],
+            le_bytes[1],
+            le_bytes[2],
+            le_bytes[3],
+        ]));
+
+        let high_bits = NanosecondsHighBits(u32::from_le_bytes([
+            le_bytes[4],
+            le_bytes[5],
+            le_bytes[6],
+            le_bytes[7],
+        ]));
+
+        (low_bits, high_bits)
+    }
+
+    /// Get the underlying value as a convenient primitive converted to seconds
+    pub fn as_secs(&self) -> u64 {
+        self.0 / Self::SECOND.0
+    }
+
+    /// Get the underlying value as a convenient primitive converted to milliseconds
+    pub fn as_millis(&self) -> u64 {
+        self.0 / Self::MILLISECOND.0
+    }
+
+    /// Get the underlying value as a convenient primitive converted to microseconds
+    pub fn as_micros(&self) -> u64 {
+        self.0 / Self::MICROSECOND.0
+    }
+}
+
+impl Default for Nanoseconds {
+    fn default() -> Self {
+        Nanoseconds::zero()
+    }
+}
+
+impl NanosecondResolution {
+    /// Unspecified resolution
+    pub const UNSPECIFIED: Self = NanosecondResolution(0);
+
+    /// Returns true if the resolution is unspecified
+    pub fn is_unspecified(&self) -> bool {
+        self.0 == Self::UNSPECIFIED.0
+    }
+}
+
+impl Default for NanosecondResolution {
+    fn default() -> Self {
+        NanosecondResolution::UNSPECIFIED
+    }
+}
+
+impl From<u32> for NanosecondResolution {
+    fn from(r: u32) -> Self {
+        NanosecondResolution(r)
+    }
+}
+
+impl From<NanosecondResolution> for u32 {
+    fn from(r: NanosecondResolution) -> Self {
+        r.0
+    }
+}
+
+impl From<u32> for NanosecondsHighBits {
+    fn from(bits: u32) -> Self {
+        NanosecondsHighBits(bits)
+    }
+}
+
+impl From<NanosecondsHighBits> for u32 {
+    fn from(bits: NanosecondsHighBits) -> Self {
+        bits.0
+    }
+}
+
+impl From<u32> for NanosecondsLowBits {
+    fn from(bits: u32) -> Self {
+        NanosecondsLowBits(bits)
+    }
+}
+
+impl From<NanosecondsLowBits> for u32 {
+    fn from(bits: NanosecondsLowBits) -> Self {
+        bits.0
+    }
+}
+
+impl WallClockId {
+    /// This time domain is local to the probe only
+    pub const fn local_only() -> Self {
```

Comment:
This one deserves a `const LOCAL_ONLY: WallClockId`, too, I think.  Maybe an `is_local_only() -> bool` too if you're feeling generous.